fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/cursors/OrElseCursor.java

```diff
@@ -57,7 +65,44 @@
     public OrElseCursor(@Nonnull RecordCursor<T> inner, @Nonnull Function<Executor, RecordCursor<T>> func) {
```

Comment:
Should this constructor be deprecated, too?

Reply:
Our various analyzers do not understand transitive dependencies, so marking this as "deprecated" creates a warning in the deprecated `RecordCursor.orElse()` method. I can suppress the warning, but I don't actually mind this constructor sticking around.

Reply:
Interesting. I thought using deprecated code in `@Deprecated` methods was generally allowed. If it's not, I think depreciating this constructor and then suppressing deprecation warnings in the deprecated `orElse` method is still the call.

Reply:
Okay, that's fine with me.